src/intern.rs

```diff
@@ -51,6 +61,9 @@ fn like_doctest_intern() {
     assert_eq!(x, Intern::from_ref("hello"));
     assert_eq!(y, Intern::from_ref("world"));
     assert_eq!(&*x, "hello"); // dereference a Intern like a pointer\
+    let _ = Intern::new("Fortunato");
+    assert!(Intern::is_interned(&"Fortunato"));
+    assert!(!Intern::is_interned(&"Montresor"));
```

Comment:
**suggestion:** Could you also test if we can pass just a `&str` to `is_interned`?

```suggestion
    assert!(Intern::is_interned(&"Fortunato"));
    assert!(!Intern::is_interned("Fortunato"));
    assert!(!Intern::is_interned(&"Montresor"));
    assert!(!Intern::is_interned("Montresor"));
```

Reply:
The problem is the && and & cases can't be unified trivially. Or I don't understand enough rust black magic to do so. 

Basically Intern::is_interned(&"") checks Intern<&str> and Intern::is_interned("") checks Intern<str> which are different.